modules/standard/Random.chpl

```diff
@@ -141,15 +141,18 @@ module Random {
     compilerError("Random.fillRandom is only defined for numeric arrays");
   }
 
-  /* Shuffle the elements of an array into a random order.
+  /* Shuffle the elements of the array into a random order.
 
-     :arg arr: a 1-D non-strided array
+     :arg arr: a rectangular 1-D non-strided array
      :arg seed: the seed to use when shuffling. Defaults to
       `oddCurrentTime` from :type:`RandomSupport.SeedGenerator`.
      :arg algorithm: A param indicating which algorithm to use. Defaults to PCG.
      :type algorithm: :type:`RNG`
    */
   proc shuffle(arr: [], seed: int(64) = SeedGenerator.oddCurrentTime, param algorithm=RNG.PCG) {
+    if(algorithm==RNG.NPB) then
+      compilerError("Cannot use NPB Random number generator for array shuffling");
```

Comment:
What's the rationale for this change?  What about it doesn't work?

Reply:
The NPBRandom module does not implement the shuffle and permutation methods. So I thought it might be a good idea to throw compileError if shuffle/permutation is invoked with the NPB Random number generator.

Reply:
Huh...  So what is the role of the `param algorithm` argument if `.shuffle` is called as a method on a RandomStream?  It seems redundant (and looks like it's currently an unused argument?  It might be useful to have @mppf weigh in on this.

Reply:
(specifically, I'm wondering what the behavior of `myPCGRandomStream.shuffle(algorithm=RNG.NPB)` or `myNPBStream.shuffle(algorithm=RNG.PCG)` would be)

Reply:
Hi @bradcray,
The implementation of permutation/shuffle in the individual modules `PCGRandom` and `NPBRandom` do not accept algorithm as a parameter. In other words what I mean to say is the methods shuffle/permutation in the Random module provide a wrapper for the corresponding methods in the individual modules `PCGRandom` and `NPBRandom`

Reply:
Ah, I think I understand better now.  I was thinking this was a method on either PCGRandom or NPBRandom, but it's actually a wrapper function that dispatches to a stream of one type or the other.  And the NPB stream doesn't support the shuffle() routine.  Thanks for clarifying.

Reply:
A better error message for cases like this might be "The NPB random number generator currently does not support array shuffling" since it could be extended to do so in the future?  (not that that's a particularly high priority).